pypesto/petab/importer.py

```diff
@@ -726,24 +738,27 @@ def create_problem(
 
         # Raise error if the correct calculator is not used.
 
-        if self._non_quantitative_data_types:
+        if self._non_quantitative_data_types is not None and any(
+            data_type in self._non_quantitative_data_types
+            for data_type in [ORDINAL, CENSORED, SEMIQUANTITATIVE]
+        ):
             if not isinstance(objective.calculator, InnerCalculatorCollector):
                 raise AssertionError(
-                    f"If there are ordinal, censored or nonlinear-monotone measurements, the `calculator` attribute of the `objective` has to be {InnerCalculatorCollector} and not {objective.calculator}."
+                    f"If there are ordinal, censored or semi-quantitative measurements, the `calculator` attribute of the `objective` has to be {InnerCalculatorCollector} and not {objective.calculator}."
                 )
-        elif self._hierarchical:
-            if not isinstance(
-                objective.calculator, HierarchicalAmiciCalculator
-            ):
+        elif (
```

Comment:
this case can be merged with the case above, right?

Reply:
Yeah, this can be much simpler. Now that relative data is merged with other data types in the Collector, it always has to be the collector if the importer is hierarchical.
```
# Raise error if the correct calculator is not used.
        if self._hierarchical:
            if not isinstance(objective.calculator, InnerCalculatorCollector):
                raise AssertionError(
                    f"If hierarchical optimization is enabled, the `calculator` attribute of the `objective` has to be {InnerCalculatorCollector} and not {objective.calculator}."
                )
```